### PR TITLE
First optimal solution attempt

### PR DESCRIPTION
**Pull request by gauravdeshmukh**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: No custom hash function used

* Execution time: ~40s on Mac M1
* Execution time of reference implementation: ~190s on Mac M1

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+            this.staticHashCode = this.stringBytes.hashCode();
+        }
+
+        public byte[] getBytes() {
+            return stringBytes.getBytes(StandardCharsets.UTF_8);
+        }
+
+        @Override
+        public boolean equals(Object bs) {
+            return bs.hashCode() == this.staticHashCode;
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java:59]: How does that handle hash collisions?

Comment (gauravdeshmukh) [reply src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java:59]: It doesn't. It relies on low probability of collision given the max size of string being 100 bytes.

Comment (gauravdeshmukh) [reply src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java]: Updated PR to remove that assumption and to use default string equals method. There are no changes in the execution time on local machine.

Comment (gauravdeshmukh) [reply src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java]: @gunnarmorling  can you please check updated PR?

**Comment by gunnarmorling** [comment]
00:16.953. Thx for participating!

**Comment by gauravdeshmukh** [comment]
@gunnarmorling is there a way to know the runtime of this code on 1brc target machine?

**Pull request merged**